modules/systemSettings.js

```diff
@@ -0,0 +1,321 @@
+  var that = this;
+  var path = ffosbr.settings.backupPaths.systemsettings + '/systemSettings.json';
+
+  //this.clean();
+
+  var settings = this.loadFromDevice();
+
+  var settingsToWrite = new File([JSON.stringify(settings)], 'systemSettings.json', {
+    type: 'text/json'
+  });
+
+  console.log(settingsToWrite);
+
+  ffosbr.media.put('sdcard1', settingsToWrite, path);
+
+};
+
+/**
+ *@description: helper function to be used in update field.
+ *  Needed to allow access to this.systemsettings
+ *
+ *@params: pass in the name of the setting (field) and 
+ *  the result from the lock.get (settingResult).
+ */
+
+SystemSettings.prototype.onsuccess = function(field, settingResult) {
+  this.systemsettings[field] = settingResult[field];
+
+
+};
+
+
+/**
+ *@description: helper function to be used in load settings. 
+ * Given one setting entry from the system settings, we acquire the lock on that setting, and
+ * copy it to our local copy of settings.
+ *@access: public
+ *@params: field is the name of the given setting
+ *
+ */
+
+SystemSettings.prototype.updateField = function(field) {
+  var lock = navigator.mozSettings.createLock();
+
+  var setting = lock.get(field);
+
+
+  setting.onsuccess = function() {
+    ffosbr.systemSettings.onsuccess(field, setting.result);
+  };
+
+  setting.onerror = function() {
+    console.warn('An error occured');
+  };
+
+};
+
+
+
+/**
+ *@description: load System Settings from device storage
+ * save these settings to a JSON object
+ *@access: public, requires 'settings' in manifest
+ */
+SystemSettings.prototype.loadFromDevice = function() {
+
+  for (var field in this.systemsettings) {
+    ffosbr.systemSettings.updateField(field);
+  }
+
+  return this.systemsettings;
+
+};
+
+
+/**
+ *@description: After the settings have been loaded, call this function
+ *  to write to local storage.
+ *
+ */
+
+SystemSettings.prototype.writeToLocalStorage = function() {
+  localStorage.setItem('ffosbrSystemSettings', JSON.stringify(this.systemsettings));
+  console.log(this.systemsettings);
+};
+
+
+/**
+ *@description: helper function to be used in write settings to device.
+ * given one setting entry, this function gets the lock on that setting and updates 
+ * the system setting based on our local copy.
+ *
+ */
+
+SystemSettings.prototype.setField = function(settingName, settingValue) {
+  var lock = navigator.mozSettings.createLock();
+  var settingString = settingName + ': ' + settingValue;
+  var settingObj = {
+    settingName: settingValue
+  };
+  var result = lock.set(settingObj);
+
+  result.onsuccess = function() {
+    console.log('successfully updated setting');
+  };
+  result.onerror = function() {
+    console.log('failed to update setting');
+  };
+};
+
+/**
+ *@description: This function first loads our backed up copy of system settings.
+ *  We then update the system settings based on what our backed up copy has stored.
+ *
+ *
+ */
+
+
+SystemSettings.prototype.writeSettingsToDevice = function() {
```

Comment:
The difference between `writeSettingsToDevice()` and `writeToLocalStorage()` is not made clear by names